Add GitHub sources in PublicAPI and CLI

frelon opened the pull request:
work in progress for implementing Git, GitHub, Gist and Http sources in CLI tool and PublicAPI

enricosada commented:
about cli, @forki @matthid what do you think?

```
paket git add .. // and list remove etc
paket github add ..
paket http add ..
```

like `git remote`?

add lot of cmd `add-git` `remove-git` etc is annoying

Asking because i'd like something like that for repotool too

frelon commented:
@enricosada I changed the cli and added the functions in AddProcess etc. Now you can run

`paket github add fsprojects/paket`

and it generates the paket.dependencies and paket.lock file correctly. It also supports the version, file and group parameters.

forki commented:
so no longer WIP?
@matthid  @enricosada can you please review?

frelon commented:
I could clean up the empty functions AddHttp, AddGit etc if you want to merge only the Github-functionality. 
Then it's no longer WIP, no.

agross commented on src/Paket/Commands.fs:
```diff
@@ -625,6 +647,7 @@ with
         member this.Usage =
             match this with
             | Add _ -> "add a new dependency"
+            | Github _ -> "commands to manipulate gihub repository references"
```
Could you please change `gihub` to `GitHub`?

frelon replied:
Absolutely! Opened #3100 for this.
